internal/tools/securitypolicy/helpers/helpers.go

```diff
@@ -137,18 +142,123 @@ func PolicyContainersFromConfigs(containerConfigs []securitypolicy.ContainerConf
 			workingDir = containerConfig.WorkingDir
 		}
 
-		container, err := securitypolicy.NewContainer(
+		container, err := CreateContainerPolicy(
 			containerConfig.Command,
 			layerHashes,
 			envRules,
 			workingDir,
 			containerConfig.ExpectedMounts,
+			containerConfig.Mounts,
 		)
 		if err != nil {
 			return nil, err
 		}
-		policyContainers = append(policyContainers, container)
+		policyContainers = append(policyContainers, &container)
 	}
 
 	return policyContainers, nil
 }
+
+// CreateContainerPolicy creates a new Container policy instance from the
+// provided constraints or an error if parameter validation fails.
+func CreateContainerPolicy(
+	command, layers []string,
+	envRuleConfigs []config.EnvRuleConfig,
+	workingDir string,
+	eMounts []string,
+	mntConfigs []config.MountConfig,
+) (securitypolicy.Container, error) {
+	envRules, err := createEnvVarRules(envRuleConfigs)
+	if err != nil {
+		return securitypolicy.Container{}, err
+	}
+	mntConstraints, err := createMountConstraints(mntConfigs)
+	if err != nil {
+		return securitypolicy.Container{}, err
+	}
+	return securitypolicy.NewContainer(
+		command,
+		layers,
+		envRules,
+		workingDir,
+		eMounts,
+		mntConstraints,
+	), nil
+}
+
+func createEnvVarRules(rules []config.EnvRuleConfig) ([]securitypolicy.EnvVarRule, error) {
+	var envRules []securitypolicy.EnvVarRule
+	for _, rule := range rules {
+		switch rule.Strategy {
+		case securitypolicy.EnvRuleRegex:
+			if _, err := regexp.Compile(rule.Rule); err != nil {
+				return nil, err
+			}
+		case securitypolicy.EnvRuleString:
+		default:
+			return nil, fmt.Errorf("invalid env_var strategy: %s", rule.Strategy)
+		}
+		envRules = append(envRules, securitypolicy.EnvVarRule{
+			Strategy: rule.Strategy,
+			Rule:     rule.Rule,
+		})
+	}
+	return envRules, nil
+}
+
+func createMountConstraints(mntConfigs []config.MountConfig) ([]securitypolicy.Mount, error) {
+	var mntConstraints []securitypolicy.Mount
+	for _, m := range mntConfigs {
+		if _, err := regexp.Compile(m.HostPath); err != nil {
+			return nil, err
+		}
+		mntConstraints = append(mntConstraints, newMountFromConfig(&m))
+	}
+	return mntConstraints, nil
+}
+
+// newOptionsFromConfig applies the same logic as CRI plugin to generate
+// mount options given readonly and propagation config.
+// TODO: (anmaxvl) update when support for other mount types is added,
+//   e.g., vhd:// or evd://
+// TODO: (anmaxvl) Do we need to set/validate Linux rootfs propagation?
```

Comment:
I think it would just fail if the rootfs propagation is not set correctly. Is that sufficient?